File: src/backend_dotnet/HRMApi/Models/MonthlyPointRule.cs
using System;

namespace HRMApi.Models;

// Quy tắc cộng điểm hàng tháng theo role
public partial class MonthlyPointRule
{
    public int Id { get; set; }
    public int RoleId { get; set; }
    public int PointValue { get; set; }
    public bool? IsActive { get; set; }
    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public virtual Role Role { get; set; } = null!;
}

File: src/backend_java/src/main/java/com/group07/human_resource_management/entity/MonthlyPointRule.java
package com.group07.human_resource_management.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "monthly_point_rules")
@Getter @Setter
@Builder @NoArgsConstructor @AllArgsConstructor
public class MonthlyPointRule {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "role_id", nullable = false, unique = true)
    private Role role;

    @Column(name = "point_value", nullable = false)
    private Integer pointValue;

    @Column(name = "is_active")
    private Boolean isActive;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
